base/dict.jl

```diff
@@ -258,12 +267,13 @@ function isequal(l::Associative, r::Associative)
 end
 
 function ==(l::Associative, r::Associative)
+    l === r && return true
     if isa(l,ObjectIdDict) != isa(r,ObjectIdDict)
         return false
     end
     if length(l) != length(r) return false end
-    for (key, value) in l
-        if value != get(r, key, secret_table_token)
+    for pair in l
+        if !in(pair, r, ==)
```

Comment:
Hmm, this change (together with the type constraint in the `in` above) makes custom `Associative`s that doesn't define their own `==` and return tuples in their iterator to raise method error in comparison. Is this intended? @vtjnash 

This causes a test failure in PyCall. c.c. @stevengj